workflow/cron/operator.go

```diff
@@ -321,3 +325,14 @@ func (woc *cronWfOperationCtx) reportCronWorkflowError(conditionType v1alpha1.Co
 	})
 	woc.metrics.CronWorkflowSubmissionError()
 }
+
+func inferScheduledRuntime() time.Time {
```

Comment:
could we get this from our cron library rather than time.now?

Reply:
That was also my initial goal. I had spent some time trying to do so, but it didn't look possible. I'll take another crack at it, because I do think it is significantly preferable. However, I don't think this solution is all too bad.

Reply:
I wonder if we do a two-part solution. This is a short term fix, giving time to investigate properly. My concern is that `time.Now()` is non-deterministic. A Go GC, or other CPU scheduling incident, could happily make the time of creation different by >1 second.

Reply:
I have devised a deterministic solution to this. I can discuss it during stand up.

> A Go GC, or other CPU scheduling incident, could happily make the time of creation different by >1 second.

Note that the margin of error in this function is not 1 second, but 59.99 seconds. As long as this is called before 59.99 seconds of the scheduled time, it will work.